diff --git a/dist/scene.js b/dist/scene.js
index e435877e54171105aee6713d08badcdfa700f0ad..2db7d7d0943cad022aad6821871c0e99ba0d49d5 100644
--- a/dist/scene.js
+++ b/dist/scene.js
@@ -1832,6 +1832,8 @@ class Xe {
   bindTexture(t, e) {
     const s = this.gl, i = e.endsWith("3D"), r = i ? s.TEXTURE_3D : s.TEXTURE_2D, n = ct(e);
     let o = this.textures.get(t);
+    const a = this.nextTextureUnit();
+    s.activeTexture(s.TEXTURE0 + a);
     if (!o || t.needsUpdate) {
       o || (o = s.createTexture(), this.textures.set(t, o)), s.bindTexture(r, o), s.texParameteri(r, s.TEXTURE_WRAP_S, s.CLAMP_TO_EDGE), s.texParameteri(r, s.TEXTURE_WRAP_T, s.CLAMP_TO_EDGE), i && s.texParameteri(r, s.TEXTURE_WRAP_R, s.CLAMP_TO_EDGE);
       const c = n ? s.NEAREST : s.LINEAR;
@@ -1854,8 +1856,7 @@ class Xe {
       } else h != null && !i && !n && s.texImage2D(s.TEXTURE_2D, 0, s.RGBA, s.RGBA, s.UNSIGNED_BYTE, h);
       t.needsUpdate = !1;
     }
-    const a = this.nextTextureUnit();
-    return s.activeTexture(s.TEXTURE0 + a), s.bindTexture(r, o), a;
+    return s.bindTexture(r, o), a;
   }
   nextTextureUnit() {
     const t = this.gl, e = t.getParameter(t.MAX_TEXTURE_IMAGE_UNITS);
